File: 800/2202A/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            int x = sc.nextInt();
            int y = sc.nextInt();
            
            int sum = (x+y);
            
            if(sum >= 0 && sum%3 == 0 && sum/3 >= Math.abs(y))
                System.out.println("YES");
            else
                System.out.println("NO");
        }
    }
}

File: 800/2202A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        int x, y;
        cin >> x >> y;

        int sum = x + y;

        if (sum >= 0 && sum % 3 == 0 && sum / 3 >= abs(y))
            cout << "YES\n";
        else
            cout << "NO\n";
    }

    return 0;
}
